components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/RegistryContentResource.java

```diff
@@ -0,0 +1,441 @@
+
+        String contentType = axis2MessageContext.getProperty(CONTENT_TYPE).toString();
+        String mediaType = Utils.getQueryParameter(messageContext, MEDIA_TYPE_KEY);
+        JSONObject jsonBody = new JSONObject();
+        if (Objects.isNull(mediaType)) {
+            mediaType = DEFAULT_MEDIA_TYPE;
+        }
+
+        MicroIntegratorRegistry microIntegratorRegistry = new MicroIntegratorRegistry();
+        if (Objects.nonNull(contentType) && contentType.contains(CONTENT_TYPE_MULTIPART_FORM_DATA)) {
+            byte[] fileContent = getPayloadFromMultipart(messageContext);
+            if (Objects.nonNull(fileContent)) {
+                microIntegratorRegistry.addMultipartResource(registryPath, mediaType, fileContent);
+                jsonBody.put("message", "Successfully added the registry resource");
+            } else {
+                jsonBody = Utils.createJsonError("Error while fetching file content from payload", axis2MessageContext,
+                        BAD_REQUEST);
+            }
+        } else if (Objects.nonNull(contentType)) {
+            String fileContent = getPayload(axis2MessageContext);
+            if (Objects.nonNull(fileContent)) {
+                microIntegratorRegistry.newNonEmptyResource(registryPath, false, mediaType, fileContent, "");
+                jsonBody.put("message", "Successfully added the registry resource");
+            } else {
+                jsonBody = Utils.createJsonError("Error while fetching file content from payload", axis2MessageContext,
+                        BAD_REQUEST);
+            }
+        } else {
+            jsonBody = Utils.createJsonError("Error while fetching Content-Type from request", axis2MessageContext,
+                    BAD_REQUEST);
+        }
+
+
+
+        return jsonBody;
+    }
+
+    /**
+     * This method handles PUT.
+     *
+     * @param messageContext      Synapse message context
+     * @param axis2MessageContext AXIS2 message context
+     */
+    private void handlePut(MessageContext messageContext, org.apache.axis2.context.MessageContext axis2MessageContext) {
+
+        String registryPath = Utils.getQueryParameter(messageContext, REGISTRY_PATH);
+        String validatedPath;
+        JSONObject jsonBody;
+        axis2MessageContext.removeProperty(Constants.NO_ENTITY_BODY);
+        if (Objects.nonNull(registryPath)) {
+            validatedPath = validatePath(registryPath, axis2MessageContext);
+            if (Objects.nonNull(validatedPath)) {
+                String pathWithPrefix = getRegistryPathPrefix(validatedPath);
+                if (!isRegistryExist(validatedPath)) {
+                    jsonBody = Utils.createJsonError("Registry does not exists in the path: " + registryPath,
+                            axis2MessageContext, BAD_REQUEST);
+                } else if (Objects.nonNull(pathWithPrefix)) {
+                    jsonBody = putRegistryArtifact(axis2MessageContext, messageContext, pathWithPrefix);
```

Comment:
Let's add audit logs for MI management modify operations. Check the following sample:

https://github.com/wso2/micro-integrator/blob/master/components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/ProxyServiceResource.java#L258